### ISSUE TITLE
Include `git` or otherwise address "current commit information was not captured by the build: git was not found in the system: ..."

### ISSUE THREAD

**Issue by eriksw** [docker-library/docker#482]
When I use the `docker` image as a build step in Google Cloud Build to perform buildx builds, I get the following warning:

    WARNING: current commit information was not captured by the build: git was not found in the system: exec: "git": executable file not found in $PATH

I presume this has to do with provenance being enabled by default per https://docs.docker.com/build/attestations/slsa-provenance/#mode

I would like to be able to use the `docker` image to run `docker buildx ...` and have the provenance information captured as designed by the new default. I know about the option to suppress the warning by using `BUILDX_NO_DEFAULT_ATTESTATIONS=1` to disable the feature entirely, but I would prefer to get it working instead.

As an end user I have no preference between what I presume are the two possible solutions:

1. Modify the provenance feature to extract the required information without running `git` as a child process.
2. Add `git` to the `docker` image.

(I'm opening this issue here because I presume the second option is much more reasonable to implement.)

**Comment by eriksw** [comment]
Additional context:

Unlike CI/CD environments like GitHub Actions where you could have an `apt-get install git` step make `git` available to later `docker buildx ...` steps, in Google Cloud Build every step (including invocation of docker) happens in a new docker container of the step's specified image, with only specific shared mounts being available to persist/pass artifacts and state between steps.

To work around the lack of `git` in the upstream `docker` image by adding it in to steps that run buildx, you'd wind up doing something equivalent to `docker run ... docker /bin/bash -c 'apt-get update; apt-get install git; docker buildx ..."` for every "run docker" step in the build definition.

**Comment by whalelines** [comment]
The `docker:git` image includes the git package. Please try using that.

**Comment by eriksw** [comment]
That does work, thank you for pointing it out.

I guess there remains an open question of whether it's appropriate for the default to not include git, given the dependency on it added upstream. In other words: should 'docker' and 'docker:git' be made the same?

**Comment by jnoordsij** [comment]
When looking through my (GitLab) build logs I ran into this and figured that #483 does not exactly solve this: it looks like this message requires the `git` binary within the `cli` variant rather than within the `dind` variant.

As an example, I set up a [very minimal GitLab repo](https://gitlab.com/jnoordsij/test-docker-ci) using a reduced version of [GitLab's example Docker CI template](https://gitlab.com/gitlab-org/gitlab-foss/-/blob/master/lib/gitlab/ci/templates/Docker.gitlab-ci.yml) to build a Hello World image. This results in the [build reporting the aforementioned warning](https://gitlab.com/jnoordsij/test-docker-ci/-/jobs/7215243387#L40), which is not present when `dind` is also used to build the image (see https://gitlab.com/jnoordsij/test-docker-ci/-/jobs/7215262223).

Therefore I was wondering:
- Is a follow-up to #483, moving the `git` install from `dind` to `cli`, appropriate? This might once again boil down to size comparison?
- Or should instead GitLab's template be updated to use `dind` both as service and image? This feels a bit off to do, but given we're already pulling in the larger `dind` image, this probably is not an issue? Or is there some `dind` entrypoint logic that might hurt performance in some way?

Note: if this should be (re-)opened and/or discussed elsewhere, let me know!
